tasties_app/models.py

```diff
@@ -1,3 +1,87 @@
-# from django.db import models
+from django.db import models
+from django.utils import timezone
+from django.contrib.auth.models import User
+from django.core.validators import MinValueValidator, MinLengthValidator, MaxValueValidator
+from datetime import datetime
 
-# Create your models here.
+
+class Category(models.Model):
+    category_name = models.CharField(max_length=16, unique=True)
+
+
+class Recipe(models.Model):
+    title = models.CharField(max_length=64, validators=[MinLengthValidator(1)], unique=True)
+    author_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    categories = models.ManyToManyField(Category)
+    description = models.CharField(max_length=512, validators=[MinLengthValidator(1)])
+    directions = models.CharField(max_length=65536, validators=[MinLengthValidator(1)])
+    publication_date = models.DateField(default=timezone.now)
+    minutes_to_make = models.IntegerField(validators=[MinValueValidator(1)])
+    recipe_picture = models.ImageField(upload_to='recipe_pictures')
+
+    def __str__(self):
+        return self.title
+
+
+class Ingredient(models.Model):
+    recipe_id = models.ForeignKey(Recipe, on_delete=models.CASCADE)
+    amount = models.DecimalField(max_digits=10, decimal_places=2, validators=[MinValueValidator(0.0001)])
+    measurement_unit = models.TextChoices('measurement_unit', 'ML FLOZ TSP OZ CUP GRAM WHOLE')
+    description = models.CharField(max_length=64, validators=[MinLengthValidator(1)])
+
+
+class Comment(models.Model):
+    author_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    recipe_id = models.ForeignKey(Recipe, on_delete=models.CASCADE)
+    publication_date = models.DateField(default=timezone.now)
+    comment_text = models.CharField(max_length=512, validators=[MinLengthValidator(1)])
+
+    def __str__(self):
+        return self.comment_text
+
+    @staticmethod
+    def show_all_comments():
+        comments = Comment.objects.all()
+        lis = []
+        for i in comments:
+            lis.append(str(i.author_id)+' '+i.comment_text)
+        return lis
+
+    def show_all_comment_author(slef):
+        a_id = slef.author_id
+        obj = Comment.objects.filter(author_id=a_id)
+        lis = []
+        for i in obj:
+            lis.append(i.comment_text)
+        return lis
+
+    def add_comment(self, text):
+        dat = datetime.today().strftime('%Y-%m-%d')
+        comm = Comment(author_id=self.author_id, recipe_id=self.recipe_id, publication_date=str(dat), comment_text=text)
+        comm.save()
+        return comm
+
+    def remove_comment(self, text):
+        a_id = self.author_id
+        comment = Comment.objects.filter(author_id=a_id, comment_text=text)
+        comment[0].delete()
+        comment2 = Comment.objects.get(author_id=a_id)
+        return comment2.comment_text
+
+    def edite_comment(self, r_id, old_comment, new_comment):
```

Comment:
First, you have a typo - the method name should be "edit_comment". This method actually should be written as an instance method. There's no reason to get take r_id and old_comment as parameters, since we have that information (from self). In this method, we don't have to do anything other than:
`def edit_comment(self, new_comment):`
`    try:`
`        self.comment_text = new_comment`
`        self.clean_fields()`
`        self.save()`
`    except ValidationError as e:`
`        raise(e)`
(but please check me, see what works)

Reply:
i make changes please check the files